### PR TITLE
creation of product CRUD

### LINKED ISSUE #28: Implementation Endpoint CRUD Product

### PR DESCRIPTION
**Pull request by nokiiam**
close #28

### CODE REVIEW

File: src/controller/ProductController.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  async one(@Param('barcode') barcode: string) {
+    const product = await this.repository.findOne(barcode);
+    if (product) {
+      return product;
+    }
+    const object : OpenFoodFactProductResponse =
+      await request(`https://fr.openfoodfacts.org/api/v0/produit/${barcode}`, { json : true });
+    const newProd : Product = { barcode: object.code, listProducts: [], promotions: [] };
+    await this.create({ barcode: object.code } as Product);
+    return newProd;
```
Comment (MrColorado) [inline src/controller/ProductController.ts]: You should be able to return the `.save()`

File: src/controller/ProductController.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return this.repository.save(products);
+  }
+
+  @Delete('/products/:barcode')
+  async remove(@Param('barcode') barcode: string) {
+    const productToRemove = await this.repository.findOne(barcode);
+    if (productToRemove) {
+      await this.repository.remove(productToRemove);
+    }
+    return productToRemove;
```
Comment (MrColorado) [inline src/controller/ProductController.ts]: It could be good to return a 404 error

File: src/controller/ProductController.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  @Patch('/products/:barcode')
+  async update(@Param('barcode') barcode: string,
+               @Body() product: Product) {
+    return this.repository.update(barcode, product);
```
Comment (MrColorado) [inline src/controller/ProductController.ts]: Look how we are updating in other controller

File: src/controller/ProductController.ts (hunk 1 above)
Comment (nokiiam) [reply src/controller/ProductController.ts]: done

File: src/controller/ProductController.ts (hunk 2 above)
Comment (nokiiam) [reply src/controller/ProductController.ts]: Done

File: src/controller/ProductController.ts (hunk 3 above)
Comment (nokiiam) [reply src/controller/ProductController.ts]: done

File: src/controller/ProductController.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  // Useless now because product only have his barcode.
+  @Patch('/products/:barcode')
+  async update(@Param('barcode') barcode: string,
+               @Body() product: Product) {
+    const existing = await this.repository.findOne(barcode);
+    if (existing === undefined) {
+      return undefined;
+    }
+    const fieldsToChange = [];
+    for (const field in fieldsToChange) {
```
Comment (MrColorado) [inline src/controller/ProductController.ts]: This will not work, the `field` variable is not the name of the field but only a int

**Comment by MrColorado** [comment]
You should include tests of this controller in this MR

File: src/controller/ProductController.ts (hunk 4 above)
Comment (nokiiam) [reply src/controller/ProductController.ts]: fixed

**Comment by nokiiam** [comment]
J'ai ajouté des petits tests

**Pull request merged**